proposals/p1983.md

```diff
@@ -0,0 +1,183 @@
+    -   [Disallow digit separators in fractions](#disallow-digit-separators-in-fractions)
+
+<!-- tocstop -->
+
+## Abstract
+
+[Proposal #143: Numeric literals](/proposals/p0143.md) added digit separators
+with strict rules for placement. It missed some use-cases. In order to address
+this, remove placement rules for numeric literals.
+
+## Problem
+
+Digit separator placement rules are too strict:
+
+-   For integers, while the original proposal mentioned Indian digit groupings,
+    Chinese, Japanese, and Korean cultures use 4-digit groupings. This oversight
+    is likely due to the description on
+    [wikipedia](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping):
+    "but the delimiter commonly separates every three digits".
+-   There are microformats where different placement rules may be desirable. See
+    [alternatives](#alternatives-considered) for more specific examples.
+
+## Background
+
+[Proposal #143: Numeric literals](/proposals/p0143.md) added digit separators
+with strict rules for placement:
+
+-   For decimal integers, the digit separators shall occur every three digits
+    starting from the right. For example, `2_147_483_648`.
+-   For hexadecimal integers, the digit separators shall occur every four digits
+    starting from the right. For example, `0x7FFF_FFFF`.
+-   For real number literals, digit separators can appear in the decimal and
+    hexadecimal integer portions (prior to the period and after the optional `e`
+    or mandatory `p`) as described in the previous bullets. For example,
+    `2_147.483648e12_345` or `0x1_00CA.FEF00Dp+24`
+-   For binary literals, digit separators can appear between any two digits. For
+    example, `0b1_000_101_11`.
+
+This was asked on
+[Issue #1485: Reconsider digit separators](https://github.com/carbon-language/carbon-lang/issues/1485),
+where a proposal was requested.
+
+## Proposal
+
+Switch to simple rules: a single digit separator can appear between any two
+digits. For example:
+
+-   Decimal literals: `1_2_3`
+-   Hexadecimal literals: `0xA_B_C`
+-   Real literals: `1_2.3_4e5_6`
+-   Binary literals are unaffected.
+
+## Rationale
+
+-   [Community and culture](/docs/project/goals.md#community-and-culture)
+    -   Removing restrictions on decimal literals provides flexibility for
+        developers who don't use 3-digit groupings.
+-   [Code that is easy to read, understand, and write](/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write)
+    -   This allows for mistakes in groupings, which is undesirable. However,
+        it's useful for microformats to be able to provide their own particular
+        groupings.
+-   [Interoperability with and migration from existing C++ code](/docs/project/goals.md#interoperability-with-and-migration-from-existing-c-code)
+    -   Reduces migration hurdles for C++ code by providing a rule that's more
+        consistent.
```

Comment:
```suggestion
    -   Reduces migration hurdles for C++ code by providing a rule that's more
        consistent with the C++ rule.
```
I assume the intended reading here is "more consistent with C++", not "more self-consistent" (the new rule is both of these things).